### PR TITLE
Issue 248 mouse events fix

### PR DESCRIPTION
**Pull request by daniel-martic-sociomantic**
Issue #248 mouse events fix

### CODE REVIEW

File: src/Module/tests.jsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ describe( 'Module', () =>
                     title        : 'hello',
                 } );
 
-                expect( wrapper.find( IconWithTooltip ).prop( 'onMouseOver' ) )
+                expect( wrapper.find( IconWithTooltip ).prop( 'onMouseEnter' ) )
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/tests.jsx]: The IconWithTooltip-related props here should remain as `onMouseOver`/`onMouseOut`

File: src/TagInput/index.jsx [hunk 2]
```diff
@@ -231,14 +231,14 @@ export default class TagInput extends Component
...
-                    className   = { cssMap.container }
-                    htmlFor     = { id }
-                    onMouseOut  = { onMouseOut }
-                    onMouseOver = { onMouseOver }
-                    style       = { { height } }>
+                    className    = { cssMap.container }
+                    htmlFor      = { id }
+                    onMouseLeave = { onMouseOut }
+                    onMouseEnter = { onMouseOver }
+                    style        = { { height } }>
```
Comment (conor-cafferkey-sociomantic) [inline src/TagInput/index.jsx]: I think we should probably get rid of the separate events on `<label>` here

File: src/TagInput/index.jsx [hunk 3]
```diff
@@ -231,14 +231,14 @@ export default class TagInput extends Component
                 isDisabled            = { isDisabled }
                 label                 = { label }
                 labelPosition         = { labelPosition }
-                onMouseOut            = { onMouseOut }
-                onMouseOver           = { onMouseOver }>
+                onMouseLeave          = { onMouseOut }
+                onMouseEnter          = { onMouseOver }>
```
Comment (conor-cafferkey-sociomantic) [inline src/TagInput/index.jsx]: These should remain as `onMouseOver`/`onMouseOut`

File: src/Module/index.jsx [hunk 4]
```diff
@@ -193,8 +193,8 @@ export default class Module extends Component
                                 message          = { errorMessage }
                                 iconType         = "error"
                                 tooltipIsVisible = { errorMessageIsVisible }
-                                onMouseOut       = { onMouseOutError }
-                                onMouseOver      = { onMouseOverError } />
+                                onMouseLeave     = { onMouseOutError }
+                                onMouseEnter     = { onMouseOverError } />
```
Comment (conor-cafferkey-sociomantic) [inline src/Module/index.jsx]: These should remain as `onMouseOver`/`onMouseOut`

File: src/CodeEditor/tests.jsx [hunk 5]
```diff
@@ -53,7 +53,7 @@ describe( 'CodeEditor', () =>
             wrapper.setProps( { onMouseOver } );
 
             expect( wrapper.find( `.${cssMap.editor}` )
-                .prop( 'onMouseOver' ) ).to.equal( onMouseOver );
+                .prop( 'onMouseEnter' ) ).to.equal( onMouseOver );
```
Comment (conor-cafferkey-sociomantic) [inline src/CodeEditor/tests.jsx]: I think we should change the test descriptions for clarity:
- `should pass onMouseOver to the container div` -> `should pass onMouseOver to the container div as onMouseEnter`
- `should pass onMouseOut to the container div` - > `should pass onMouseOut to the container div as onMouseLeave`

File: src/NavItem/index.jsx [hunk 6]
```diff
@@ -95,7 +95,7 @@ NavItem.propTypes =
     /*
     *  Item represents the current page and/or section
      */
-    isCurrent     : PropTypes.bool,
+    isCurrentPage : PropTypes.bool,
```
Comment (daniel-martic-sociomantic) [inline src/NavItem/index.jsx]: This one was specified as prop, but was missing in props validation (there's also props validation for `isCurrent` )

**Pull request merged**